lib/ui/window.dart

```diff
@@ -719,85 +195,157 @@ class Window {
+  /// [SceneBuilder.addPicture]. With the [SceneBuilder.build] method you can
+  /// then obtain a [Scene] object, which you can display to the user via this
+  /// [render] function.
   ///
   /// See also:
   ///
-  ///  * [WidgetsBindingObserver], for a mechanism at the widgets layer to
-  ///    register for notifications when this is called.
-  ///  * [MediaQuery.of], a simpler mechanism for the same.
-  VoidCallback? get onMetricsChanged => _onMetricsChanged;
-  VoidCallback? _onMetricsChanged;
-  Zone _onMetricsChangedZone = Zone.root;
+  ///  * [SchedulerBinding], the Flutter framework class which manages the
+  ///    scheduling of frames.
+  ///  * [RendererBinding], the Flutter framework class which manages layout and
+  ///    painting.
+  void render(Scene scene) => platformDispatcher.render(scene, this);
+}
+
+/// A top-level platform window displaying a Flutter layer tree drawn from a
+/// [Scene].
+///
+/// The current list of all Flutter views for the application is available from
+/// `WidgetsBinding.instance.platformDispatcher.views`. Only views that are of
+/// type [FlutterWindow] are top level platform windows.
+///
+/// There is also a [PlatformDispatcher.instance] singleton object in `dart:ui`
+/// if `WidgetsBinding` is unavailable, but we strongly advise avoiding a static
+/// reference to it. See the documentation for [PlatformDispatcher.instance] for
+/// more details about why it should be avoided.
+///
+/// See also:
+///
+///  * [PlatformDispatcher], which manages the current list of [FlutterView]
+///    (and thus [FlutterWindow]) instances.
+class FlutterWindow extends FlutterView {
+  FlutterWindow._(this._windowId, this.platformDispatcher);
+
+  /// The opaque ID for this view.
+  final Object _windowId;
+
+  @override
+  final PlatformDispatcher platformDispatcher;
+
+  /// The configuration of this view.
+  @override
+  ViewConfiguration get viewConfiguration {
+    assert(platformDispatcher._viewConfigurations.containsKey(_windowId));
+    return platformDispatcher._viewConfigurations[_windowId] ?? const ViewConfiguration();
+  }
+}
+
+/// A [FlutterWindow] that includes access to setting callbacks and retrieving
+/// properties that reside on the [PlatformDispatcher].
+///
+/// It is the type of the `WidgetsBinding.instance.window` singleton and the
+/// legacy global [window] singleton.
+///
+/// This class provides backward compatibility with code that was written before
+/// Flutter supported multiple top level windows. To modify or retrieve these
+/// properties, new code should use `WidgetsBinding.instance.platformDispatcher`.
+///
+/// There is also a [PlatformDispatcher.instance] singleton object in `dart:ui`
+/// if `WidgetsBinding` is unavailable. But we strongly advise avoiding a static
+/// reference to it. See the documentation for [PlatformDispatcher.instance] for
+/// more details about why it should be avoided.
+class SingletonFlutterWindow extends FlutterWindow {
```

Comment:
Is the plan to deprecate this and `WidgetsBinding.instance.window` at some point? Or will this be around forever?

Also, should we call this the `LegacyFlutterWindow` to make clear that it's just around for historic reasons?